src/AppInstallerRepositoryCore/Microsoft/ARPHelper.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "Microsoft/SQLiteIndex.h"
+#include <AppInstallerArchitecture.h>
+#include <winget/Registry.h>
+#include <winget/ManifestInstaller.h>
+#include <wil/resource.h>
+
+#include <string>
+
+namespace AppInstaller::Repository::Microsoft
+{
+    // A helper to find the various locations that contain ARP (Add/Remove Programs) entries.
+    struct ARPHelper
+    {
+        // See https://docs.microsoft.com/en-us/windows/win32/msi/uninstall-registry-key for details.
+        std::wstring SubKeyPath{ L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall" };
+
+        // REG_SZ
+        std::wstring DisplayName{ L"DisplayName" };
+        // REG_SZ
+        std::wstring Publisher{ L"Publisher" };
+        // REG_SZ
+        std::wstring DisplayVersion{ L"DisplayVersion" };
+        // REG_DWORD (ex. 0xMMmmbbbb, M[ajor], m[inor], b[uild])
+        std::wstring Version{ L"Version" };
+        // REG_DWORD
+        std::wstring VersionMajor{ L"VersionMajor" };
+        // REG_DWORD
+        std::wstring VersionMinor{ L"VersionMinor" };
+        // REG_SZ
+        std::wstring URLInfoAbout{ L"URLInfoAbout" };
+        // REG_SZ
+        std::wstring HelpLink{ L"HelpLink" };
+        // REG_SZ
+        std::wstring InstallLocation{ L"InstallLocation" };
+        // REG_DWORD (ex. 1033 [en-us])
+        std::wstring Language{ L"Language" };
+        // REG_SZ (ex. "english")
+        std::wstring InnoSetupLanguage{ L"Inno Setup: Language" };

```

Comment:
>InnoSetupLanguage [](start = 21, length = 17)

Didn't know inno has its own arp registry entry. Do other types we support have their own entries?

Reply:
Then should we get input from other installer packaging providers to build these index helper functions as they could have different way to store online updater info, dependencies?

Reply:
@yao-msft , I did not see any others, but this was mostly from poking around my own machine's registry.  It is not necessarily a complete list (and I don't even use this currently).  I don't even know if this value is actually relevant to the package's language, or is just the language used during installation.

@quangkieu , agreed.  There are most likely other values that exist, and as we expand our support we will need to know them.  Do you have any in mind currently?  I can add a note.